tests/vm_state_test.go

```diff
@@ -46,12 +43,12 @@ var _ = Describe("[sig-storage]VM state", decorators.SigStorage, decorators.Requ
 		startVM := func(vm *v1.VirtualMachine) {
 			By("Starting the VM")
 			vm = libvmops.StartVirtualMachine(vm)
-			vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, k8smetav1.GetOptions{})
+			vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
 			By("Ensuring the firmware is done so we don't send any keystroke to it")
-			err = console.LinuxExpecter(vmi)
-			Expect(err).ToNot(HaveOccurred())
+			// Wait for cloud init to finish and start the agent inside the vmi.
+			Eventually(matcher.ThisVMI(vmi), 4*time.Minute, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceAgentConnected))
```

Comment:
nit: don't use Eventually

Reply:
What else? Booting the VMI will take some time.

Reply:
I think the request was to use WithPolling/WithTimeout, changed :+1: